File: n485-max-consecutive-ones.go
package main

func findMaxConsecutiveOnes(nums []int) int {
	max := 0
	cons := 0

	for i := 0; i < len(nums); i++ {
		if nums[i] == 1 {
			cons++
			if cons > max {
				max = cons
			}
		} else {
			cons = 0
		}
	}

	return max
}

File: n485-max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxv = 0
        cons = 0
        
        for i in xrange(len(nums)):
            if nums[i] == 1:
                cons += 1
                maxv = max(maxv, cons)
            else:
                cons = 0
        
        return maxv
